spring-integration-hazelcast/src/test/java/org/springframework/integration/hazelcast/inbound/HazelcastClusterMonitorInboundChannelAdapterTests.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.hazelcast.inbound;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.GroupConfig;
+import com.hazelcast.core.Client;
+import com.hazelcast.core.ClientType;
+import com.hazelcast.core.DistributedObjectEvent;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import com.hazelcast.core.LifecycleEvent;
+import com.hazelcast.core.LifecycleEvent.LifecycleState;
+import com.hazelcast.core.MembershipEvent;
+import com.hazelcast.core.MigrationEvent;
+
+/**
+ * Hazelcast Cluster Monitor Inbound Channel Adapter Unit Test Class
+ *
+ * @author Eren Avsarogullari
+ * @since 1.0.0
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration
+@DirtiesContext
+public class HazelcastClusterMonitorInboundChannelAdapterTests {
+
+	private static final String TEST_GROUP_NAME1 = "Test_Group_Name1";
+
+	private static final String TEST_GROUP_NAME3 = "Test_Group_Name3";
+
+	private static final int TIMEOUT = 2_000;
```

Comment:
This timeout isn't enough for `async` tests. The `testMigrationEvent` fails for me constantly.
So, 10 seconds would be better.

From other side this test class takes 1 min and more.
I understand that the problem with cluster team building and leader election, but can't we somehow decrease the number of `HazelcastInstance`s to get better performance?
Thanks